plugins/channel_track.go

```diff
@@ -196,7 +208,81 @@ func (p *ChannelTracker) nickCallback(b *seabird.Bot, m *irc.Message) {
 	//fmt.Printf("%s (%s) changed their name to %s\n", oldUser, p.uuids[newUser], newUser)
 }
 
-func (p *ChannelTracker) namesCallback(b *seabird.Bot, m *irc.Message) {
+func (p *ChannelTracker) modeCallback(b *seabird.Bot, m *irc.Message) {
+	// We only care about MODE messages where a specific user is
+	// changed.
+	if len(m.Params) < 3 {
+		return
+	}
+
+	logger := b.GetLogger()
+
+	channel := m.Params[0]
+	target := m.Params[2]
+
+	//change := m.Params[1]
+	//op := change[0]
+	//mode := change[1]
+
+	// Ensure we know about this user and this channel
+	u := p.LookupUser(target)
+	c := p.LookupChannel(channel)
+	if u == nil || c == nil {
+		logger.Warnf("Got MODE callback for %s on %s but we aren't tracking both", target, channel)
+		return
+	}
+
+	// Just send a WHO request and clear out the modes for this user because
+	// mode parsing is hard.
+	u.channels[channel] = make(map[rune]bool)
+	b.Writef("WHO :%s", target)
+}
+
+func (p *ChannelTracker) whoCallback(b *seabird.Bot, m *irc.Message) {
+	// Filter out broken messages
+	if len(m.Params) < 7 {
+		return
+	}
+
+	prefixes, ok := p.getSymbolToPrefixMapping(b)
+	if !ok {
+		return
+	}
+
+	var (
+		channel = m.Params[0]
+		//user    = m.Params[1]
```

Comment:
I'm keeping these in for now, because I'd like to be able to come back to this and track more things later without worrying about re-looking up these params.